frontend/src/screens/SingleProduct/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import useCustomContext from '../../customHooks/Hook';
+import { PRODUCTS_DETAILS_REQUEST, PRODUCTS_DETAILS_SUCCESS, PRODUCTS_DETAILS_FAILURE } from '../../constants/ProductConstants';
+import axios from 'axios';
+import { ADD_TO_CART } from '../../constants/CartConstants';
+import { Loader, Message, Rating } from '../../components'
+import style from './singleproduct.module.css';
+import { Link } from 'react-router-dom';
+
+
+const SingleProduct = ({ match: { params: { id } } }) => {
+
+    const { state, dispatch } = useCustomContext();
+    const { product, loading, error, cart } = state;
+    const [qty, setQty] = useState(1);
+
+    const fetchSingleProductDetails = async () => {
+        try {
+            dispatch({ type: PRODUCTS_DETAILS_REQUEST });
+            const { data } = await axios.get(`/api/product/${id}`);
+            dispatch({ type: PRODUCTS_DETAILS_SUCCESS, payload: data.products })
+        } catch (err) {
+            dispatch({ type: PRODUCTS_DETAILS_FAILURE, payload: 'Something went wrong' })
+        }
+    }
+
+    useEffect(() => {
+        fetchSingleProductDetails()
+    }, [])
+
+    const addToCart = () => {
+        const updateQtyProduct = { ...product, qty: qty }
+        dispatch({ type: ADD_TO_CART, payload: updateQtyProduct })
+    }
+
+    const checkincart = () => {
+        return !!cart.find((x) => x.id === product.id)
+    }
+
+    const { name, description, countInStock, image, rating, numReviews, category, price } = product;
+
+    return (
+        <div className={style.single_product_container}>
+            <Link to={{ pathname: '/products' }} className="p-half"> <button className="btn btn-primary">Go Back</button> </Link>
+            {
+                loading ? <Loader /> : error ? <Message variant='danger'>{error}</Message> : (
+
+                    <div className={style.single_product_card}>
+
+                        <div className={style.single_product_image_container}>
+                            <img src={image} alt="single-product-image" className={style.single_product_image} />
+                        </div>
+
+                        <div className={style.single_product_details_container}>
+                            <ul className="list-group">
+                                <li className={style.list_item}><strong>{name}</strong></li>
+                                <li className={style.list_item}>{description}</li>
+                                <li className={style.list_item}>{countInStock > 0 ? 'In Stock' : 'Out of Stock'}</li>
+                                <li className={style.list_item}><Rating numReviews={numReviews} value={rating} /></li>
+                                <li className={style.list_item}>
+                                    <select value={qty} onChange={(e) => setQty((qty) => Number(e.target.value))} className="select-css">
+                                        {
```

Comment:
check for count before proceeding, else can throw warning error

Reply:
If countInStock is zero then it wont show anything and we cannot add that to cart because button gets disabled